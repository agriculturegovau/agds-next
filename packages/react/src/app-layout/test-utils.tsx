import { Fragment } from 'react';
import { Text } from '../text';
import { NotificationBadge } from '../notification-badge';
import { VisuallyHidden } from '../a11y';
import {
	ChartLineIcon,
	EmailIcon,
	ExitIcon,
	HelpIcon,
	HomeIcon,
	SuccessIcon,
	SettingsIcon,
	FactoryIcon,
	AvatarIcon,
	ChevronsLeftIcon,
	MinusIcon,
} from '../icon';
import {
	DropdownMenuDivider,
	DropdownMenuItem,
	DropdownMenuItemLink,
	DropdownMenuPanel,
	DropdownMenuGroup,
	DropdownMenuGroupLink,
	DropdownMenuItemRadio,
} from '../dropdown-menu';

export const navigationItems = (businessName: string) => [
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: 'Back to my account',
				icon: ChevronsLeftIcon,
				href: '/account',
			},
		],
	},
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: (
					<Fragment>
						<Text fontWeight="bold" fontSize="xs">
							{businessName}
						</Text>
						<Text color="muted" fontSize="xs">
							ABN: 00 000 000 000
						</Text>
					</Fragment>
				),
			},
		],
	},
	[
		{
			label: 'Dashboard',
			icon: HomeIcon,
			href: '/',
		},
		{
			label: 'Establishments',
			icon: FactoryIcon,
			href: '/establishments',
			items: [
				{
					label: 'Canberra',
					icon: MinusIcon,
					href: '/establishments/canberra',
				},
				{
					label: 'Sydney',
					icon: MinusIcon,
					href: '/establishments/sydney',
				},
			],
		},
		{
			label: 'Data and Insights',
			icon: ChartLineIcon,
			href: '/intelligence',
		},
		{
			label: 'Compliance',
			icon: SuccessIcon,
			href: '/compliance',
			items: [
				{
					label: 'Audit',
					icon: MinusIcon,
					href: '/compliance/audit',
				},
				{
					label: 'Certificates',
					icon: MinusIcon,
					href: '/compliance/certificates',
				},
			],
		},
	],

	[
		{
			label: 'Help',
			icon: HelpIcon,
			href: '/help',
		},
	],
	[
		{
			label: 'Sign out',
			onClick: console.log,
			icon: ExitIcon,
		},
	],
];

export const exampleData = {
	userNames: {
		short: 'Ab Cd',
		regular: 'Toto Wolff',
		medium: 'Benjamin Harrington',
		long: 'Benjamin Alexander Harrington-Smythe',
	},
	businessNames: {
		short: ['Lorem', 'Ipsum', 'Company'],
		regular: ['Antfix', 'Produce Fresh', 'Organic Co'],
		medium: [
			'Malesuada lacus a tortor blandit fermentum phasellus',
			'Iaculis tortor duis ante nec risus elementum id ui',
			'Purus tortor lacus malesuada phasellus ipsum ex duis libero ante id',
		],
		long: [
			'Duis quis iaculis libero nec consectetur luctus ex in malesuada lacus a tortor blandit fermentum phasellus iaculis ipsum nec purus aliquet id elementum risus duis ante ui',
			'Duis ut a libero duis nisl ex lacus in luctus viverra iaculis iaculis tortor duis malesuada blandit purus diam ipsum ante nec risus elementum id ui nec quis aliquet phasellus fermentum',
			'Blandit iaculis iaculis quis ante diam viverra elementum ui risus nec luctus purus tortor lacus malesuada phasellus ipsum ex duis libero ante id Duis fermentum nisl aliquet duis in a nec ut consectetur',
		],
	},
};

export function ExampleAccountDropdown({
	businesses = exampleData.businessNames.regular,
	onBusinessChange,
	selectedBusinessName,
}: {
	businesses?: string[];
	onBusinessChange: (value: string) => void;
	selectedBusinessName: string;
}) {
	return (
		<DropdownMenuPanel palette="light">
			<DropdownMenuGroup label="Businesses">
				{businesses.map((businessName) => (
					<DropdownMenuItemRadio
						key={businessName}
						checked={businessName === selectedBusinessName}
						secondaryText="ABN 00 000 000 000"
						onClick={() => onBusinessChange?.(businessName)}
					>
						{businessName}
					</DropdownMenuItemRadio>
				))}
				<DropdownMenuGroupLink href="#">View all</DropdownMenuGroupLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuGroup label="My account">
				<DropdownMenuItemLink href="/profile" icon={AvatarIcon}>
					Profile
				</DropdownMenuItemLink>
				<DropdownMenuItemLink
					href="/messages"
					icon={EmailIcon}
					endElement={
						<span>
							<NotificationBadge tone="action" value={6} max={99} aria-hidden />
							<VisuallyHidden>, 6 unread</VisuallyHidden>
						</span>
					}
				>
					Messages
				</DropdownMenuItemLink>
				<DropdownMenuItemLink href="/account-settings" icon={SettingsIcon}>
					Account settings
				</DropdownMenuItemLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuItem onClick={() => console.log('sign out')} icon={ExitIcon}>
				Sign out
			</DropdownMenuItem>
		</DropdownMenuPanel>
	);
}
