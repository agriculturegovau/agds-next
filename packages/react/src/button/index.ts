'use client';
export * from './Button';
export * from './ButtonGroup';
export * from './BaseButton';
export * from './styles';
