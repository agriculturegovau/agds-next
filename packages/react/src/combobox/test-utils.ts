export type Option = { label: string; value: string };

export const COUNTRY_OPTIONS = [
	'Afghanistan is a really long country that spans multiple lines cmon this can be way longer than that Afghanistan is a really long country that spans multiple lines cmon this can be way longer than that',
	'Albania',
	'Algeria',
	'Andorra',
	'Angola',
	'Antigua and Barbuda',
	'Argentina',
	'Armenia',
	'Australia',
	'Austria',
	'Azerbaijan',
	'Bahamas',
	'Bahrain',
	'Bangladesh',
	'Barbados',
	'Belarus',
	'Belgium',
	'Belize',
	'Benin',
	'Bhutan',
	'Bolivia',
	'Bosnia and Herzegovina',
	'Botswana',
	'Brazil',
	'Brunei',
	'Bulgaria',
	'Burkina Faso',
	'Burundi',
	"CÃ´te d'Ivoire",
	'Cabo Verde',
	'Cambodia',
	'Cameroon',
	'Canada',
	'Central African Republic',
	'Chad',
	'Chile',
	'China',
	'Colombia',
	'Comoros',
	'Congo (Congo-Brazzaville)',
	'Costa Rica',
	'Croatia',
	'Cuba',
	'Cyprus',
	'Czechia (Czech Republic)',
	'Democratic Republic of the Congo',
	'Denmark',
	'Djibouti',
	'Dominica',
	'Dominican Republic',
	'Ecuador',
	'Egypt',
	'El Salvador',
	'Equatorial Guinea',
	'Eritrea',
	'Estonia',
	'Eswatini (fmr. "Swaziland")',
	'Ethiopia',
	'Fiji',
	'Finland',
	'France',
	'Gabon',
	'Gambia',
	'Georgia',
	'Germany',
	'Ghana',
	'Greece',
	'Grenada',
	'Guatemala',
	'Guinea',
	'Guinea-Bissau',
	'Guyana',
	'Haiti',
	'Holy See',
	'Honduras',
	'Hungary',
	'Iceland',
	'India',
	'Indonesia',
	'Iran',
	'Iraq',
	'Ireland',
	'Israel',
	'Italy',
	'Jamaica',
	'Japan',
	'Jordan',
	'Kazakhstan',
	'Kenya',
	'Kiribati',
	'Kuwait',
	'Kyrgyzstan',
	'Laos',
	'Latvia',
	'Lebanon',
	'Lesotho',
	'Liberia',
	'Libya',
	'Liechtenstein',
	'Lithuania',
	'Luxembourg',
	'Madagascar',
	'Malawi',
	'Malaysia',
	'Maldives',
	'Mali',
	'Malta',
	'Marshall Islands',
	'Mauritania',
	'Mauritius',
	'Mexico',
	'Micronesia',
	'Moldova',
	'Monaco',
	'Mongolia',
	'Montenegro',
	'Morocco',
	'Mozambique',
	'Myanmar (formerly Burma)',
	'Namibia',
	'Nauru',
	'Nepal',
	'Netherlands',
	'New Zealand',
	'Nicaragua',
	'Niger',
	'Nigeria',
	'North Korea',
	'North Macedonia',
	'Norway',
	'Oman',
	'Pakistan',
	'Palau',
	'Palestine State',
	'Panama',
	'Papua New Guinea',
	'Paraguay',
	'Peru',
	'Philippines',
	'Poland',
	'Portugal',
	'Qatar',
	'Romania',
	'Russia',
	'Rwanda',
	'Saint Kitts and Nevis',
	'Saint Lucia',
	'Saint Vincent and the Grenadines',
	'Samoa',
	'San Marino',
	'Sao Tome and Principe',
	'Saudi Arabia',
	'Senegal',
	'Serbia',
	'Seychelles',
	'Sierra Leone',
	'Singapore',
	'Slovakia',
	'Slovenia',
	'Solomon Islands',
	'Somalia',
	'South Africa',
	'South Korea',
	'South Sudan',
	'Spain',
	'Sri Lanka',
	'Sudan',
	'Suriname',
	'Sweden',
	'Switzerland',
	'Syria',
	'Tajikistan',
	'Tanzania',
	'Thailand',
	'Timor-Leste',
	'Togo',
	'Tonga',
	'Trinidad and Tobago',
	'Tunisia',
	'Turkey',
	'Turkmenistan',
	'Tuvalu',
	'Uganda',
	'Ukraine',
	'United Arab Emirates',
	'United Kingdom',
	'United States of America',
	'Uruguay',
	'Uzbekistan',
	'Vanuatu',
	'Venezuela',
	'Vietnam',
	'Yemen',
	'Zambia',
	'Zimbabwe',
].map((country) => ({ label: country, value: country }));

export const STATE_OPTIONS = [
	{ label: 'Australian Capital Territory', value: 'act' },
	{ label: 'New South Wales', value: 'nsw' },
	{ label: 'Northern Territory', value: 'nt' },
	{ label: 'Queensland', value: 'qld' },
	{ label: 'South Australia', value: 'sa' },
	{ label: 'Tasmania', value: 'tas' },
	{ label: 'Victoria', value: 'vic' },
	{ label: 'Western Australia', value: 'wa' },
];

export const NAME_OPTIONS = [
	{
		fullName: 'John Doe',
		jobTitle: 'Product Manager',
		status: 'Draft',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Jane Smith',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 4,
	},
	{
		fullName: 'Robert Johnson',
		jobTitle: 'UX Designer',
		status: 'Deactivated',
		unreadMessageCount: 5,
	},
	{
		fullName: 'Emily Davis',
		jobTitle: 'Product Manager',
		status: 'Approved',
		unreadMessageCount: 6,
	},
	{
		fullName: 'Michael Williams',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Sarah Wilson',
		jobTitle: 'UX Designer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'David Brown',
		jobTitle: 'Product Manager',
		status: 'Deactivated',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Olivia Taylor',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 6,
	},
	{
		fullName: 'James Moore',
		jobTitle: 'UX Designer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Sophia Clark',
		jobTitle: 'Product Manager',
		status: 'Approved',
		unreadMessageCount: 0,
	},
].map((option) => ({
	...option,
	label: option.fullName,
	value: option.fullName,
}));
