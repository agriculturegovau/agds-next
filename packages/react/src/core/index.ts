export * from './Core';
export * from './CoreProvider';
export * from './tokens';
export * from './packs';
export * from './utils';
export * from './theme';
export * from './goldTheme';
export * from './boxPalette';
export * from './actionPalette';
export * from './globalPalette';
export * from './responsive';
