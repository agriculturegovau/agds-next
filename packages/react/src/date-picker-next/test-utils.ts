import * as yup from 'yup';

export const errorMessage = 'Enter a valid date';

// `yup.date()` can sometimes give false positives with certain string values
// Fixes https://github.com/jquense/yup/issues/764
// Tests for this can be viewed in the tests for `DatePicker` and `DateRangePicker`
// IMPORTANT: This is for example purposes only. It should NOT be exported from the design system
export const yupDateField = yup
	.date()
	.transform((current, orig) => (typeof orig === 'string' ? orig : current))
	.typeError(errorMessage);
