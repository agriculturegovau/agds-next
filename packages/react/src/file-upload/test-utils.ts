import { FileStatus, FileWithStatus } from './utils';

export function createExampleFileWithStatus(status?: FileStatus) {
	const bits = ['example'];
	const file: FileWithStatus = new File(bits, 'example.txt', {
		type: 'text/plain',
	});
	file.status = status;
	return file;
}

export function createExampleImageFileWithStatus(status?: FileStatus) {
	const file: FileWithStatus = new File(['example'], 'example.jpg', {
		type: 'image/jpg',
	});
	file.status = status;
	return file;
}
