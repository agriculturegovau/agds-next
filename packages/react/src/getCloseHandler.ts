// eslint-disable-next-line @typescript-eslint/no-explicit-any
type Fn = (params?: any) => void;

export const closeHandlerWarningMessage =
	'Do not use onDismiss in combination with onClose. Use onClose by itself. onDismiss is deprecated. Only onClose will be called.';

export function getRequiredCloseHandler(
	onClose: Fn | undefined,
	onDismiss: Fn | undefined
): Fn {
	const closeHandler = onClose ? onClose : onDismiss;

	handleWarnings(onClose, onDismiss);

	if (!closeHandler) {
		throw new Error('onClose prop is required');
	}

	return closeHandler;
}

export function getOptionalCloseHandler(
	onClose: Fn | undefined,
	onDismiss: Fn | undefined
): Fn | undefined {
	handleWarnings(onClose, onDismiss);

	return onClose ? onClose : onDismiss;
}

function handleWarnings(
	onClose: Fn | undefined,
	onDismiss: Fn | undefined
) {
	if (onClose && onDismiss) {
		console.warn(closeHandlerWarningMessage)
	} else if (onDismiss) {
		console.warn('onDismiss is deprecated. Use onClose instead.')
	}
}
