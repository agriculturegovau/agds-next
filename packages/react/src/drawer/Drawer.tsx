import {
	Fragment,
	FunctionComponent,
	MouseEventHandler,
	useEffect,
	useRef,
} from 'react';
import { Global } from '@emotion/react';
import { createPortal } from 'react-dom';
import { useTransition, animated, SpringValue } from '@react-spring/web';
import {
	boxPalette,
	canUseDOM,
	tokens,
	useAriaModalPolyfill,
	usePrefersReducedMotion,
} from '../core';
import { getRequiredCloseHandler } from '../getCloseHandler';
import { DrawerDialog } from './DrawerDialog';
import { type DrawerProps } from './types';

export const Drawer: FunctionComponent<DrawerProps> = ({
	actions,
	children,
	elementToFocusOnClose,
	isOpen = false,
	mutedOverlay = false,
	onClose,
	onDismiss,
	title,
	width = 'md',
}) => {
	const handleClose = getRequiredCloseHandler(onClose, onDismiss);
	const scrollbarWidth = useRef<number>(0);

	useEffect(() => {
		scrollbarWidth.current =
			window.innerWidth - document.documentElement.clientWidth;
	}, []);

	// Close the Drawer when the user presses the escape key
	useEffect(() => {
		const handleKeyDown = (e: KeyboardEvent) => {
			if (isOpen && e.code === 'Escape') {
				e.preventDefault();
				e.stopPropagation();
				handleClose();
			}
		};
		window.addEventListener('keydown', handleKeyDown);
		return () => window.removeEventListener('keydown', handleKeyDown);
	}, [isOpen, handleClose]);

	// Polyfill usage of `aria-modal`
	const { modalContainerRef } = useAriaModalPolyfill(isOpen);

	// Animation styles
	const prefersReducedMotion = usePrefersReducedMotion();
	const dialogTransitions = useTransition([isOpen], {
		from: { translateX: '100%', opacity: 0 },
		enter: { translateX: '0%', opacity: 1 },
		leave: { translateX: '100%', opacity: 0 },
		config: { duration: 150 },
		immediate: prefersReducedMotion,
	});

	// Since react portals can not be rendered on the server and this component is always closed by default
	// This component doesn't need to be server side rendered
	if (!canUseDOM()) return null;

	return createPortal(
		<Fragment>
			{isOpen ? <LockScroll scrollbarWidth={scrollbarWidth.current} /> : null}
			{dialogTransitions(({ translateX, opacity }, item) =>
				item ? (
					<div ref={modalContainerRef}>
						<Overlay
							onClick={handleClose}
							style={{ opacity }}
							mutedOverlay={mutedOverlay}
						/>
						<DrawerDialog
							actions={actions}
							elementToFocusOnClose={elementToFocusOnClose}
							onClose={handleClose}
							style={{ translateX }}
							title={title}
							width={width}
						>
							{children}
						</DrawerDialog>
					</div>
				) : null
			)}
		</Fragment>,
		document.body
	);
};

function Overlay({
	onClick,
	style,
	mutedOverlay,
}: {
	onClick: MouseEventHandler<HTMLDivElement>;
	style: { opacity: SpringValue<number> };
	mutedOverlay: boolean;
}) {
	return (
		<animated.div
			onClick={onClick}
			css={{
				position: 'fixed',
				inset: 0,
				backgroundColor: mutedOverlay
					? boxPalette.overlayMuted
					: boxPalette.overlay,
				zIndex: tokens.zIndex.overlay,
			}}
			style={style}
		/>
	);
}

function LockScroll({ scrollbarWidth }: { scrollbarWidth: number }) {
	return (
		<Global
			styles={{
				// Fixes this Chrome/Edge bug https://issues.chromium.org/issues/40835364
				html: { marginRight: scrollbarWidth, scrollbarGutter: 'auto' },
				body: { overflow: 'hidden' },
			}}
		/>
	);
}
