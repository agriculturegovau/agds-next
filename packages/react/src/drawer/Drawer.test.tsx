import '@testing-library/jest-dom';
import 'html-validate/jest';
import { axe, toHaveNoViolations } from 'jest-axe';
import userEvent from '@testing-library/user-event';
import { useTernaryState } from '../core';
import { Button } from '../button';
import { Text } from '../text';
import { render, screen, cleanup, waitFor } from '../../../../test-utils';
import { closeHandlerErrorMessage } from '../getCloseHandler';
import { Drawer } from './Drawer';

expect.extend(toHaveNoViolations);

afterEach(cleanup);

function renderBaseDrawer() {
	return render(
		<Drawer
			isOpen
			onClose={() => undefined}
			title="Drawer title"
			actions={<Button>Example action</Button>}
		>
			<Text as="p">Example content.</Text>
		</Drawer>
	);
}

function DrawerExample() {
	const [isOpen, open, close] = useTernaryState(false);
	return (
		<div>
			<Button onClick={open} data-testid="open-button">
				Open
			</Button>
			<Drawer
				isOpen={isOpen}
				onClose={close}
				title="Drawer title"
				actions={
					<Button onClick={close} data-testid="close-button">
						Close
					</Button>
				}
			>
				<Text as="p">This is the Drawer content.</Text>
			</Drawer>
		</div>
	);
}
function OnDismissDrawerExample() {
	const [isOpen, open, close] = useTernaryState(false);
	return (
		<div>
			<Button onClick={open} data-testid="open-button">
				Open
			</Button>
			<Drawer
				isOpen={isOpen}
				onDismiss={close}
				title="Drawer title"
				actions={
					<Button onClick={close} data-testid="close-button">
						Close modal
					</Button>
				}
			>
				<Text as="p">This is the Drawer content.</Text>
			</Drawer>
		</div>
	);
}

function renderDrawer() {
	return render(<DrawerExample />);
}
function renderOnDismissDrawer() {
	return render(<OnDismissDrawerExample />);
}

describe('Drawer', () => {
	it('renders correctly', () => {
		const { baseElement } = renderBaseDrawer();
		expect(baseElement).toMatchSnapshot();
	});

	it('renders valid HTML with no a11y violations', async () => {
		const { baseElement } = renderBaseDrawer();
		expect(baseElement).toHTMLValidate({
			extends: ['html-validate:recommended'],
			rules: {
				'no-inline-style': 'off',
				// react 18s `useId` break this rule
				'valid-id': 'off',
			},
		});
		expect(await axe(baseElement)).toHaveNoViolations();
	});

	it('focuses the correct elements when opening and closing', async () => {
		renderDrawer();

		// Open the Drawer by clicking the "Open" button
		await userEvent.click(screen.getByTestId('open-button'));
		expect(await screen.findByRole('dialog')).toBeInTheDocument();

		// Title should have focus
		expect(await screen.getByText('Drawer title')).toHaveFocus();

		// Close the Drawer by clicking the "Close" button
		await userEvent.click(screen.getByTestId('close-button'));

		// After closing the Drawer, the "Open" button should be focused
		// Note: We need to wait for the closing animation
		await waitFor(() =>
			expect(screen.getByTestId('open-button')).toHaveFocus()
		);
	});
	it('onDismiss draw focuses the correct elements when opening and closing', async () => {
		renderOnDismissDrawer();

		// Open the Drawer by clicking the "Open" button
		await userEvent.click(screen.getByTestId('open-button'));
		expect(await screen.findByRole('dialog')).toBeInTheDocument();

		// Title should have focus
		expect(await screen.getByText('Drawer title')).toHaveFocus();

		// Close the Drawer by clicking the "Close" button
		await userEvent.click(screen.getByTestId('close-button'));

		// After closing the Drawer, the "Open" button should be focused
		// Note: We need to wait for the closing animation
		await waitFor(() =>
			expect(screen.getByTestId('open-button')).toHaveFocus()
		);
	});
	it('errors if provided both onClose and onDismiss', () => {
		expect(() =>
			render(
				<Drawer
					isOpen
					onClose={jest.fn}
					onDismiss={jest.fn}
					title="Draw title"
				/>
			)
		).toThrow(closeHandlerErrorMessage);
	});
});
