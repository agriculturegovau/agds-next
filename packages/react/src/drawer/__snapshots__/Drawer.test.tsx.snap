// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer renders correctly 1`] = `
<body>
  <div
    aria-hidden="true"
  />
  <div>
    <div
      class="css-hyp6kw-Overlay"
      style="opacity: 0;"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="drawer-:r0:-title"
        aria-modal="true"
        class="css-bugcec-boxStyles-DrawerDialog"
        role="dialog"
        style="transform: translateX(100%);"
      >
        <div
          class="css-1fzj2hl-boxStyles-DrawerHeader"
        >
          <h2
            class="css-17ywrtq-boxStyles-Text-DrawerHeaderTitle"
            data-autofocus="true"
            id="drawer-:r0:-title"
            tabindex="-1"
          >
            Drawer title
          </h2>
        </div>
        <div
          class="css-1r9s54p-boxStyles-DrawerContent"
        >
          <p
            class="css-6j1v6l-boxStyles-Text"
          >
            Example content.
          </p>
        </div>
        <div
          class="css-10pjjkw-boxStyles-DrawerFooter"
        >
          <button
            class="css-ysvto2-BaseButton-Button"
            type="button"
          >
            <span>
              <span
                class="css-oobk4c-Button"
              >
                Example action
              </span>
              <span
                aria-live="assertive"
              />
            </span>
          </button>
        </div>
        <button
          class="css-1yjf0d9-BaseButton-DrawerDialog"
          type="button"
        >
          <span>
            <span
              class="css-oobk4c-Button"
            >
              Close
            </span>
            <span
              aria-live="assertive"
            />
          </span>
          <svg
            aria-hidden="true"
            class="css-10mbdw9-Button"
            clip-rule="evenodd"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="18"
              x2="6"
              y1="6"
              y2="18"
            />
            <line
              x1="6"
              x2="18"
              y1="6"
              y2="18"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;
